### PR TITLE
gitlab - rearange the script for clearer output

### PR DESCRIPTION
**Pull request by jeremylt**
Our GitLab CI output is very long. This is an effort to do a bit of minor rearranging to help me look through the logs.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -15,29 +15,37 @@ noether-rocm:
...
-# LIBXSMM v1.16.1
+# -- LIBXSMM v1.16.1
     - cd .. && export XSMM_VERSION=libxsmm-1.16.1 && { [[ -d $XSMM_VERSION ]] || { git clone --depth 1 --branch 1.16.1 https://github.com/hfp/libxsmm.git $XSMM_VERSION && make -C $XSMM_VERSION -j$(nproc); }; } && git -C $XSMM_VERSION describe --tags && export XSMM_DIR=$PWD/$XSMM_VERSION && cd libCEED
-# OCCA v1.1.0
+# -- OCCA v1.1.0
     - cd .. && export OCCA_VERSION=occa-1.1.0 OCCA_OPENCL_ENABLED=0 && { [[ -d $OCCA_VERSION ]] || { git clone --depth 1 --branch v1.1.0 https://github.com/libocca/occa.git $OCCA_VERSION && make -C $OCCA_VERSION -j$(nproc); }; } && make -C $OCCA_VERSION info && export OCCA_DIR=$PWD/$OCCA_VERSION && cd libCEED
 # libCEED
     - make configure HIP_DIR=/opt/rocm OPT='-O -march=native -ffp-contract=fast'
     - BACKENDS_CPU=$(make info-backends | grep -o '/cpu[^ ]*') && BACKENDS_GPU=$(make info-backends | grep -o '/gpu[^ ]*')
+    - make info
```
Comment (jeremylt) [inline .gitlab-ci.yml:28]: Not sure when this was accidentally removed, but I find it useful in the output

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -15,29 +15,37 @@ noether-rocm:
...
     - echo '[{"subject":"/","metrics":[{"name":"Transfer Size (KB)","value":"19.5","desiredSize":"smaller"},{"name":"Speed Index","value":0,"desiredSize":"smaller"},{"name":"Total Score","value":92,"desiredSize":"larger"},{"name":"Requests","value":4,"desiredSize":"smaller"}]}]' > performance.json
     - make -k -j$NPROC_CPU BACKENDS="$BACKENDS_CPU" junit realsearch=%
     - make -k -j$NPROC_GPU BACKENDS="$BACKENDS_GPU" junit realsearch=%
-# MFEM v4.2
+# Libraries for examples
+# -- PETSc with HIP (minimal)
+    - export PETSC_DIR=/projects/jed/petsc PETSC_ARCH=mpich-hip-g && git -C $PETSC_DIR describe
+    - make -C $PETSC_DIR info
+    - make -k -j$NPROC_CPU BACKENDS="$BACKENDS_CPU" junit search="petsc fluids solids"
+    - make -k -j$NPROC_GPU BACKENDS="$BACKENDS_GPU" junit search="petsc fluids solids"
```
Comment (jedbrown) [inline .gitlab-ci.yml:39]: Isn't the `realsearch=%` above going to have run all of these tests, so you'll just be running them again here? Maybe once is enough?

Comment (jeremylt) [reply .gitlab-ci.yml:39]: The `realsearch` won't pick up on the PETSc tests until the `PETSC_DIR` is set. As it currently is, we don't get any feedback from GPU tests until after the core library tests + PETSc tests successfully run on CPU, but I feel like running the core library tests on CPU and GPU first before starting the PETSc tests gives us more information.

Comment (jedbrown) [reply .gitlab-ci.yml:39]: Okay, maybe make a comment to that effect. `PETSC_DIR` is in the default environment on Noether, but gitlab-runner must clear it out.

Maybe now would be a good time to move to `PETSC_DIR=/projects/petsc PETSC_ARCH=mpich-hip` so I can use my `/project/jed/petsc` as a sandbox.

Comment (jeremylt) [reply .gitlab-ci.yml:46]: added a comment to that effect and switched the dir/arch

Comment (jeremylt) [reply .gitlab-ci.yml:45]: naturally, there is some small change in PETSc causing a build problem

Comment (jeremylt) [reply .gitlab-ci.yml:45]: Yep, this commit last week made changes like this
https://gitlab.com/petsc/petsc/-/commit/45480ffeba491aca5d3f3b8c78679069fb317d32
```
  ierr = DMAddBoundary(dm, DM_BC_ESSENTIAL, "wall", "marker", 0, 0, NULL, (void (*)(void)) trig_u, NULL, 1, &id, user);CHKERRQ(ierr);
  ierr = DMGetLabel(dm, "marker", &label);CHKERRQ(ierr);
  ierr = DMAddBoundary(dm, DM_BC_ESSENTIAL, "wall", label, 1, &id, 0, 0, NULL, (void (*)(void)) trig_u, NULL, user, NULL);CHKERRQ(ierr);
```

But unfortunately this is not an easy fix that maintains backwards compatibility with a macro like we have seen before.

Comment (knepley) [reply .gitlab-ci.yml:45]: Alternatively, you can change to
```
  ierr = DMAddBoundaryByName(dm, DM_BC_ESSENTIAL, "wall", "marker", 0, 0, NULL, (void (*)(void)) trig_u, NULL, 1, &id, user);CHKERRQ(ierr);
```

Comment (jeremylt) [reply .gitlab-ci.yml:45]: I might have figured out a way. We'll see what it does in the Azure run, but it should work in Noether. It works locally.

Comment (jeremylt) [reply .gitlab-ci.yml:45]: Since Azure isn't on a PETSc release, we'd need to update there (or drop it if we prefer? Noether offers the same coverage)

Comment (jeremylt) [reply .gitlab-ci.yml:45]: Confirmed - Noether likes the new macro. We just have to fix Azure

**Comment by jeremylt** [comment]
These three PRs are good on my end, as feel free to merge once they're ready on yours

**Review by jedbrown — APPROVED** [review]

**Pull request merged**